src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs

```diff
@@ -17,13 +17,18 @@ public class RedisCacheOptions : IOptions<RedisCacheOptions>
         /// The configuration used to connect to Redis.
         /// </summary>
         public string Configuration { get; set; }
-        
+
         /// <summary>
         /// The configuration used to connect to Redis.
         /// This is preferred over Configuration.
         /// </summary>
         public ConfigurationOptions ConfigurationOptions { get; set; }
 
+        /// <summary>
+        /// Gets or sets a delegate to create the ConnectionMultiplexer instance.
+        /// </summary>
+        public Func<IConnectionMultiplexer> ConnectionMultiplexerFactory { get; set; }
```

Comment:
Should this be nullable?

Reply:
```suggestion
        [DisallowNull]
        public Func<Task<IConnectionMultiplexer>>? ConnectionMultiplexerFactory { get; set; }
```

Reply:
What making this a `Task` return type buying us exactly?

Reply:
@davidfowl mentioned creating the connection involved I/O. This was meant to reflect that.

Reply:
The Func will likely include an async call e.g. `ConnectionMultiplexer.ConnectAsync` and this way, the Func can be awaited in other components to get the same connection.

Reply:
@JunTaoLuo and @pranavkm I am having some issues with this. Nullable annotations does not appear to be enabled in this class yet:

```/Users/dev-spv/aspnetcore/src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs(32,65): error CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. [/Users/dev-spv/aspnetcore/src/Caching/StackExchangeRedis/src/Microsoft.Extensions.Caching.StackExchangeRedis.csproj] ```

I am willing to add these annotations if desired, but I had a ton of issues trying to satisfy the Roslyn Unshipped API file changes with VSCode.  Specifically it would not recognize the get as public even though it recognized it was part of the API. Below is what I could not get working even when I did have `#nullable enabled` at the top of the class:

`Microsoft.Extensions.Caching.StackExchangeRedis.RedisCacheOptions.ConnectionMultiplexerFactory.get -> System.Func<System.Threading.Tasks.Task<StackExchange.Redis.IConnectionMultiplexer>>?
Microsoft.Extensions.Caching.StackExchangeRedis.RedisCacheOptions.ConnectionMultiplexerFactory.set -> void`

Reply:
I'll take a look.

Reply:
Let's continue the conversation at https://github.com/dotnet/aspnetcore/pull/32266